### ISSUE TITLE
Incontinuous Sync progress on Grafana dashboard and warning on alpha 2 (bug report)

### ISSUE THREAD

**Issue by 0xDaizz** [paradigmxyz/reth#3676]
### Describe the bug

![image](https://github.com/paradigmxyz/reth/assets/90135051/9d87aaf9-842b-4c0e-ab7a-b25b7320f547)

after updating my syncing node to 0.1.0-alpha.2, my grafana dashboard got bug.

I thought it was due to the log of previous syncing status so it started from scratch, but It still gives me abnormal graphs.

seems that the syncing progress graph turns into "0" value by default and soars randomly to the current syncing block height.

and of course, the stage checkpoints does not work as well.

btw, the reth itself does not have any error. the logs of reth says the node is syncing very well(except the progress is very slow   :( well that's my hardware issue maybe).

### Steps to reproduce

[github gist tutorial](https://gist.github.com/teebaumcrypto/5c7a30ae9f25d3f628100188149b1fb1)

I used the instruction just like the link above.

```
git clone https://github.com/paradigmxyz/reth
cd reth
git checkout v0.1.0-alpha.2
RUSTFLAGS="-C target-cpu=native" cargo build --profile maxperf --features jemalloc
```

and ran my node by this command
```
RUST_LOG=info ~/reth/target/maxperf/reth node --datadir ~/data/reth_data --authrpc.jwtsecret ~/data/jwt.hex --ws --ws.addr="127.0.0.1" --ws.api=eth,web3,net,txpool --http --http.api=eth,web3,net,txpool --http.addr="127.0.0.1" --http.port=8545 --metrics 127.0.0.1:9001
```

### Node logs

```text
I could find out the only warning in the logs.


$ RUST_LOG=info ~/reth/target/maxperf/reth node --datadir ~/data/reth_data --authrpc.jwtsecret ~/data/jwt.hex --ws --ws.addr="127.0.0.1" --ws.api=eth,web3,net,txpool --http --http.api=eth,web3,net,txpool --http.addr="127.0.0.1" --http.port=8545 --metrics 127.0.0.1:9001


2023-07-09T02:28:37.169590Z  INFO reth::cli: reth 0.1.0-alpha.2 (f3d7988f) starting
2023-07-09T02:28:37.170168Z  INFO reth::cli: Configuration loaded path="/home/hw/data/reth_data/reth.toml"
2023-07-09T02:28:37.170178Z  INFO reth::cli: Opening database path="/home/hw/data/reth_data/db"
2023-07-09T02:28:37.176444Z  INFO reth::cli: Database opened
2023-07-09T02:28:37.176454Z  INFO reth::cli: Starting metrics endpoint addr=127.0.0.1:9001
2023-07-09T02:28:37.377748Z  INFO reth::cli: Pre-merge hard forks (block based):
- Frontier                         @0
- Homestead                        @1150000
- Dao                              @1920000
- Tangerine                        @2463000
- SpuriousDragon                   @2675000
- Byzantium                        @4370000
- Constantinople                   @7280000
- Petersburg                       @7280000
- Istanbul                         @9069000
- MuirGlacier                      @9200000
- Berlin                           @12244000
- London                           @12965000
- ArrowGlacier                     @13773000
- GrayGlacier                      @15050000
Merge hard forks:
- Paris                            @58750000000000000000000 (network is not known to be merged)

Post-merge hard forks (timestamp based):
- Shanghai                         @1681338455

2023-07-09T02:28:37.387059Z  INFO reth::cli: Transaction pool initialized
2023-07-09T02:28:37.387106Z  INFO reth::cli: Connecting to P2P network
2023-07-09T02:28:37.387696Z  INFO net::peers: Loading saved peers file=/home/hw/data/reth_data/known-peers.json
2023-07-09T02:28:37.388438Z  INFO reth::cli: Connected to P2P network peer_id=0x23ff…6e6c local_addr=0.0.0.0:30303
2023-07-09T02:28:37.388589Z  WARN consensus::engine: Pipeline sync progress is inconsistent first_stage_checkpoint=17650936 inconsistent_stage_id=Bodies inconsistent_stage_checkpoint=14980000
2023-07-09T02:28:37.388605Z  INFO reth::cli: Consensus engine initialized
2023-07-09T02:28:37.388613Z  INFO reth::cli: Engine API handler initialized
2023-07-09T02:28:37.389409Z  INFO reth::cli: RPC auth server started url=127.0.0.1:8551
2023-07-09T02:28:37.389629Z  INFO reth::cli: RPC IPC server started url=/tmp/reth.ipc
2023-07-09T02:28:37.389640Z  INFO reth::cli: RPC HTTP server started url=127.0.0.1:8545
2023-07-09T02:28:37.389646Z  INFO reth::cli: RPC WS server started url=127.0.0.1:8546
2023-07-09T02:28:37.389654Z  INFO reth::cli: Starting consensus engine
2023-07-09T02:28:37.389878Z  INFO reth::cli: Status connected_peers=0 latest_block=0
2023-07-09T02:28:37.390469Z  INFO reth::node::events: Executing stage pipeline_stages=1/13 stage=Headers from=17650936 checkpoint=100.0% eta=unknown
2023-07-09T02:28:37.390913Z  INFO execute{stage=Headers}: sync::stages::headers: Target block already reached checkpoint=100.0% target=Hash(0x5edc7a724b056df3d9c44f6f0108b55abdecdcb468cc4a1421619658e0978703)
2023-07-09T02:28:37.390995Z  INFO reth::node::events: Stage finished executing pipeline_stages=1/13 stage=Headers block=17650936 checkpoint=100.0% eta=unknown
2023-07-09T02:28:37.391149Z  INFO reth::node::events: Executing stage pipeline_stages=2/13 stage=TotalDifficulty from=17650936 checkpoint=100.0% eta=unknown
2023-07-09T02:28:37.391177Z  INFO reth::node::events: Stage finished executing pipeline_stages=2/13 stage=TotalDifficulty block=17650936 checkpoint=100.0% eta=unknown
2023-07-09T02:28:37.391265Z  INFO downloaders::bodies: Downloading bodies 14980001..=17650936 distance=2670935
2023-07-09T02:28:37.391261Z  INFO reth::node::events: Executing stage pipeline_stages=3/13 stage=Bodies from=14980000 checkpoint=84.9% eta=unknown
```

### Platform(s)

Linux (x86)

### What version/commit are you on?

$ ./target/maxperf/reth --version reth Version: 0.1.0-alpha.2 Commit SHA: f3d7988f Build Timestamp: 2023-07-08T18:24:08.394088990Z Build Features: jemalloc

### What database version are you on?

```
$ ./target/maxperf/reth db version
Current database version: 1
Local database is uninitialized
```

### If you've built Reth from source, provide the full command you used

``` git clone https://github.com/paradigmxyz/reth cd reth git checkout v0.1.0-alpha.2 RUSTFLAGS="-C target-cpu=native" cargo build --profile maxperf --features jemalloc ```  and ran my node by this command ``` RUST_LOG=info ~/reth/target/maxperf/reth node --datadir ~/data/reth_data --authrpc.jwtsecret ~/data/jwt.hex --ws --ws.addr="127.0.0.1" --ws.api=eth,web3,net,txpool --http --http.api=eth,web3,net,txpool --http.addr="127.0.0.1" --http.port=8545 --metrics 127.0.0.1:9001 ```

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by 0xDaizz** [comment]
Just dowwngraded to alpha 1, and the dashboard works just fine. I can assert that the problem is from the alpha 2 version.

**Comment by 0xDaizz** [comment]
![image](https://github.com/paradigmxyz/reth/assets/90135051/fe0dda91-4ea0-438f-beef-1411baa38b9c)

Here's the difference between alpha 2 and 1.

after downgrading to alpha 1, I can see the fluctuating sync block height graph got fixed, as well as the sync checkpoints barchart.

**Comment by mattsse** [comment]
ref #3670

**Comment by shekhirin** [comment]
Also ref https://github.com/paradigmxyz/reth/pull/3671

**Comment by 0xDaizz** [comment]
Resolved the bug after updating to 0.1.0-alpha.3.
